opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -480,7 +478,25 @@ public boolean hasAnnotation(File file) {
      * @return collection of added repositories
      */
     public Collection<RepositoryInfo> addRepositories(File[] files) {
-        return addRepositories(files, env.getNestingMaximum(), 0, false);
+        ExecutorService executor = env.getIndexerParallelizer().getRepositorySearchExecutor();
```

Comment:
May I suggest just using the `-T,--threads` executor? It's not otherwise being used at this stage of a run

Reply:
yes, that's reasonable.